### PR TITLE
Feature/evaluators

### PR DESCRIPTION
**Pull request by KarolinaMiq**

### CODE REVIEW

File: src/draive/evaluators/text_coherence.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from draive.evaluation import evaluator
+from draive.generation import generate_model
+from draive.parameters import DataModel
+from draive.types.multimodal import MultimodalContent, TextContent
+
+__all__ = [
+    "CoherenceScore",
```
Comment (KaQuMiQ) [inline src/draive/evaluators/text_coherence.py:7]: we don't want to export those score symbols, keep it private for all evaluators

File: src/draive/evaluators/text_coherence.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+based on the Evaluation Criteria."""
+
+@evaluator(name="coherence", threshold=0.75)
+async def text_coherence_evaluator(
+    reference: str,
+    text: str,
+    /,
+) -> float:
+    input_content = MultimodalContent.of(
+        TextContent(text=f"Reference text: {reference}\n\nDerived text: {text}") # type: ignore
```
Comment (KaQuMiQ) [inline src/draive/evaluators/text_coherence.py:45]: please never ignore typing, if there is an issue here please resolve it however there is no need to use "MultimodalContent" just use a string

Comment (KaQuMiQ) [reply src/draive/evaluators/text_coherence.py:45]: it seems that it is a common thing, please fix all occurrences

File: src/draive/evaluators/text_keywords.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from draive.evaluation import evaluator
+
+__all__ = [
+    "text_keywords_evaluator",
+]
+
+@evaluator(name="keywords", threshold=0.95)
+async def text_keywords_evaluator(text: str, keywords: list[str]) -> float:
```
Comment (KaQuMiQ) [inline src/draive/evaluators/text_keywords.py:8]: please add comma after the last argument

File: src/draive/evaluators/text_similarity.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Evaluation Steps:
+1. Read both texts carefully.
+2. Compare the semantic meaning of the texts.
+3. Assign a similarity score from 0 to 1, where higher scores indicate greater similarity.
+"""  # noqa: E501
+
+@evaluator(name="similarity", threshold=0.75)
+async def text_similarity_evaluator(
+    text1: str,
```
Comment (KaQuMiQ) [inline src/draive/evaluators/text_similarity.py:37]: I guess one of those is reference and the second one is checked/compared, please rename those arguments

File: src/draive/evaluators/text_similarity.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    model: SimilarityScore = await generate_model(
+        SimilarityScore,
+        instruction=INSTRUCTION,
+        input=input_content,
+    )
+
+    return model.score
+
+@evaluator(name="vector similarity", threshold=0.75)
+async def vector_similarity_evaluator(
```
Comment (KaQuMiQ) [inline src/draive/evaluators/text_similarity.py:53]: it is still a text evaluation yet name does not indicate this, please rename i.e. "text_vector_similarity_evaluator"

File: src/draive/evaluators/text_keywords.py (hunk 3 above)
Comment (piotr-rarus) [reply src/draive/evaluators/text_keywords.py:8]: Such behavior should be enforced by the formatter. Usually it's only enabled for multi-line.

File: src/draive/evaluators/text_coherence.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    /,
+) -> float:
+    input_content = f"Reference text: {reference}\n\nCompered text: {compared}"
+    model: CoherenceScore = await generate_model(
+        CoherenceScore,
+        instruction=INSTRUCTION,
+        input=input_content,
+    )
+
+    return model.score/5
```
Comment (piotr-rarus) [inline src/draive/evaluators/text_coherence.py:49]: Magic number 😔

Create constants:

```py
LOWER_RANGE = 1
UPPER_RANGE = 5
```
```py
return model.score/UPPER_RANGE
```

Plug those values into `INSTRUCTION`.

Comment (KaQuMiQ) [reply src/draive/evaluators/text_coherence.py:56]: we can't really extract those constant as the scale is explained within the prompt, we could not change them anyway

**Pull request merged**